src/systems/ec/implementations/vanilla/packages/ec-component-gallery/ec-component-gallery.js

```diff
@@ -408,6 +515,31 @@ export class Gallery {
     }
   }
 
+  /**
+   * Invoke listeners for on click events on view all.
+   *
+   * @param {Event} e
+   */
+  handleClickOnViewAll(e) {
```

Comment:
I noticed that this handler is the same as the standard handler for the click on an element, couldn't we re-use that by simply calling it from `handleClickOnViewAll()`?

Reply:
it isn't 100% the same. The only difference is that the view all button displays the first item in any case (`this.updateOverlay(this.galleryItems[0]);`) and the item click depends on the item selected (`this.updateOverlay(e.currentTarget);`)